services/harmony/app/frontends/health.ts

```diff
@@ -21,46 +24,79 @@ interface ComponentStatus {
 interface HealthStatus {
   status: String;
   message?: String;
-  components: ComponentStatus[];
+  dependencies: DependencyStatus[];
 
 }
 
 /**
- * Returns the health information
+ * Returns the db health information
  *
- * @param tx - the database transaction
- * @returns a promise resolving to the health check response
+ * @returns a promise resolving to the DependencyStatus of database
  */
-async function getGeneralHealth(tx: Transaction): Promise<HealthStatus> {
+async function getDbHealth(): Promise<DependencyStatus> {
   let dbHealthy = true;
   try {
-    await Job.getTimeOfMostRecentlyUpdatedJob(tx);
+    await Job.getTimeOfMostRecentlyUpdatedJob(db);
   } catch (e) {
     logger.error(e);
     dbHealthy = false;
   }
 
-  let health;
   if (dbHealthy) {
+    return {
+      name: 'db',
+      status: UP,
+    };
+  } else {
+    return {
+      name: 'db',
+      status: DOWN,
+      message: 'Unable to query the database',
+    };
+  }
+}
+
+/**
+ * Returns the health information
+ *
+ * @param context - the request context
+ * @returns a promise resolving to the health check response
+ */
+async function getGeneralHealth(context: RequestContext): Promise<HealthStatus> {
+  const dbHealth = await getDbHealth();
```

Comment:
nit: I _think_ it is slightly more efficient to not await each request, instead, gather them all in a `Promise.all` and `await` that. Probably not a big deal for this use case, though.

Reply:
Refactored.